lib/command_runner.dart

```diff
@@ -108,26 +108,26 @@ class CommandRunner<T> {
   ///
   /// This always returns a [Future] in case the command is asynchronous. The
   /// [Future] will throw a [UsageException] if [args] was invalid.
-  Future<T> run(Iterable<String> args) =>
-      Future.sync(() => runCommand(parse(args)));
+  Future<T?> run(Iterable<String> args) =>
+      Future.sync(() => runCommand(parse(args)!));
 
   /// Parses [args] and returns the result, converting an [ArgParserException]
   /// to a [UsageException].
   ///
   /// This is notionally a protected method. It may be overridden or called from
   /// subclasses, but it shouldn't be called externally.
-  ArgResults parse(Iterable<String> args) {
+  ArgResults? parse(Iterable<String> args) {
     try {
       return argParser.parse(args);
     } on ArgParserException catch (error) {
       if (error.commands.isEmpty) usageException(error.message);
 
       var command = commands[error.commands.first];
       for (var commandName in error.commands.skip(1)) {
-        command = command.subcommands[commandName];
+        command = command!.subcommands[commandName];
       }
 
-      command.usageException(error.message);
+      command!.usageException(error.message);
       return null;
```

Comment:
If we make the return type of `usageException` `Never` can we drop the `return null`?

Reply:
https://github.com/dart-lang/args/pull/160